### PR TITLE
Draw material patterns and colors on lithology chart

### PR DESCRIPTION
**Pull request by danielnaab**
Before making a pull request
----------------------------

- [x] Make sure all tests run
- [x] Update the changelog appropriately

Description
-----------
Draw material patterns and colors on lithology chart

After making a pull request
---------------------------
- [x] If appropriate, put the link to the PR in the JIRA ticket
- [x] Assign someone to review unless the change is trivial

### CODE REVIEW

File: assets/src/patterns/605.svg [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+     xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" xmlns:a="http://ns.adobe.com/AdobeSVGViewerExtensions/3.0/"
+     x="0px" y="0px" width="53.4px" height="53.2px" viewBox="0 0 53.4 53.2" style="enable-background:new 0 0 53.4 53.2;"
+     xml:space="preserve">
+<style type="text/css">
+    .st0{fill:none;stroke:#000000;stroke-width:0.7;stroke-linecap:round;}
+    .st1{fill:none;stroke:#000000;stroke-width:0.3;}
+</style>
+<defs>
+</defs>
+<line class="st0" x1="28.4" y1="9.6" x2="28.4" y2="9.6"/>
```
Comment (danielnaab) [inline assets/src/patterns/605.svg]: These Illustrator paths are problematic with the minifier - it's a zero-length line, that shows up as a single dot. Shape to path conversion (the bulk of SVG minification) ignores them. We'll need to modify in Illustrator to fix.

File: server/ngwmn/templates/macros/components.html [hunk 2]
```diff
@@ -1,4 +1,13 @@
-{% macro Graph(agency_code, site_no) -%}
+{# This component will request a remote resource and then insert it as a child into the DOM #}
+{% macro SVGPattern(url, id) %}
+    <div class="ngwmn-component" data-component="svg-pattern" data-url="{{ url }}" data-id="{{ id }}"></div>
```
Comment (danielnaab) [inline server/ngwmn/templates/macros/components.html]: This component inserts an external SVG document into the DOM in a form usable as a fill pattern.

**Comment by mbucknell** [comment]
What is assets/src/images/logo.sketch used for?

**Comment by danielnaab** [comment]
> What is assets/src/images/logo.sketch used for?

Hmm. I didn't intend to check that in - it's an old thing that's not used.

**Comment by mbucknell** [comment]
Sorry about putting comments here, but the file view is to painful.

On line 54 in assets/src/scripts/components/graph/view/construction.js should remove commented out code.

**Comment by mbucknell** [comment]
In /assets/src/scripts/components/graph/view/lithology.js my IDE complains that the imports are not used (lines 1 and 3).

**Comment by mbucknell** [comment]
This is not something that needs to dealt with at the moment, but the code is set up right now to have the lithology/construction chart next to the water-levels. If we decide to separate these two things would the code changes be relatively straightforward? Because I don't see them called out separately in the markup, I think the answer is no, but I may be missing something.

**Comment by danielnaab** [comment]
> This is not something that needs to dealt with at the moment, but the code is set up right now to have the lithology/construction chart next to the water-levels. If we decide to separate these two things would the code changes be relatively straightforward? Because I don't see them called out separately in the markup, I think the answer is no, but I may be missing something.

It wouldn't be trivial, but it'd be doable.

**Comment by mbucknell** [comment]
How hard is it to add tests for lithology_parse.parse_material? I was really hoping to see some tests so I could understand the function better.

Also I am not finding a reference for using the & operator on sets. Am I misinterpreting this? Should the union method be used instead? (line 309 in server/ngwmn/services/lithology_parser.py

**Comment by danielnaab** [comment]
> How hard is it to add tests for lithology_parse.parse_material? I was really hoping to see some tests so I could understand the function better.

Yes I can add something for that, or maybe remove the function and call the classify functions directly. Will make it clearer...

> Also I am not finding a reference for using the & operator on sets. Am I misinterpreting this? Should the union method be used instead? (line 309 in server/ngwmn/services/lithology_parser.py

Yes `&` on sets is the same as the union method.

**Pull request merged**